components/notebook-controller/pkg/controller/notebook/notebook_controller.go

```diff
@@ -283,6 +284,14 @@ func generateStatefulSet(instance *v1alpha1.Notebook) *appsv1.StatefulSet {
 			},
 		},
 	}
+
+	// Inject GCP credentials
+	if os.Getenv("GCP_CREDENTIALS") != "" {
```

Comment:
How about generalizing this?

Take an environment variable that is a comma separated list of labels to apply to the pods.

Reply:
done